src/pybind/mgr/smb/handler.py

```diff
@@ -837,6 +849,32 @@ def _check_share(
         raise ErrorResult(
             share, msg="path is not a valid directory in volume"
         )
+    if earmark_resolver:
+        earmark = earmark_resolver.get_earmark(
+            volpath,
+            share.cephfs.volume,
+        )
+        if not earmark:
+            smb_earmark = (
+                f"{EarmarkTopScope.SMB.value}.cluster.{share.cluster_id}"
+            )
+            earmark_resolver.set_earmark(
+                volpath,
+                share.cephfs.volume,
+                smb_earmark,
+            )
+        else:
+            if not earmark_resolver.check_earmark(earmark, EarmarkTopScope.SMB):
+                raise ErrorResult(
+                    share,
+                    msg=f"earmark is already been set by {earmark.split('.')[0]}",
+                )
+            # Check if earmark is set by same cluster
+            if earmark.split('.')[2] != share.cluster_id:
+                raise ErrorResult(
+                    share,
+                    msg=f"earmark is already been set by cluster {earmark.split('.')[2]}",
```

Comment:
grammar: same as above `is -> has`
since we're changing things still it might be good to say "smb cluster <x>" here to make it clear that it has been earmarked for smb but by the different cluster.